### ISSUE TITLE
lxd cannot start on Alpine

### ISSUE THREAD

**Issue by lePereT** [lxc/lxd#4141]
# Required information

 * Distribution: Alpine
 * Distribution version: 3.7
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.9.73-0-virthardened
   * LXC version:
   * LXD version: 2.21
   * Storage backend in use: dir

```
config: {}
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- macaroon_authentication
- network_sriov
- console
- restrict_devlxd
- migration_pre_copy
- infiniband
- maas_network
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
```

# Issue description

Cannot start a new container on Alpine.

# Steps to reproduce

 1. Install LXD from Alpine repos
 2. Take steps to setup LXD:

adduser rich lxd
adduser root lxd

set /etc/subuid and /etc/subgid to:

lxd:100000:65536
root:100000:65536

(The following need to be run every boot to start unprivileged containers)

service lxd start
sudo chown root:lxd /var/lib/lxd/unix.socket

 3.

lxc launch images:alpine/3.7

# Information to attach

 - [ ] Any relevant kernel output (`dmesg`)

None

 - [ ] Container log (`lxc info NAME --show-log`)

```
Name: settled-quagga
Remote: unix://
Architecture: x86_64
Created: 2018/01/06 13:09 UTC
Status: Stopped
Type: persistent
Profiles: default
```

Log:

```
            lxc 20180106131213.548 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
            lxc 20180106131213.548 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
            lxc 20180106131213.577 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2122 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2122 0 100000 65536
            lxc 20180106131213.577 ERROR    lxc_start - start.c:lxc_spawn:1276 - Failed to set up id mapping.
            lxc 20180106131213.577 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
            lxc 20180106131213.577 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
            lxc 20180106131213.577 ERROR    lxc_container - lxccontainer.c:wait_on_daemonized_start:760 - Received container state "ABORTING" instead of "RUNNING"
            lxc 20180106131213.577 ERROR    lxc_start - start.c:__lxc_start:1459 - Failed to spawn container "settled-quagga".
            lxc 20180106131213.577 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
            lxc 20180106131213.577 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
            lxc 20180106131213.579 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2125 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2125 0 100000 65536 65536 0 1
            lxc 20180106131213.579 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2125"
            lxc 20180106131213.581 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2128 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2128 0 100000 65536 65536 0 1
            lxc 20180106131213.581 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2128"
            lxc 20180106131213.582 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2131 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2131 0 100000 65536 65536 0 1
            lxc 20180106131213.582 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2131"
            lxc 20180106131213.583 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2134 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2134 0 100000 65536 65536 0 1
            lxc 20180106131213.583 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2134"
            lxc 20180106131213.587 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2137 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2137 0 100000 65536 65536 0 1
            lxc 20180106131213.587 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2137"
            lxc 20180106131213.589 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2140 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2140 0 100000 65536 65536 0 1
            lxc 20180106131213.589 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2140"
            lxc 20180106131213.590 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2143 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2143 0 100000 65536 65536 0 1
            lxc 20180106131213.590 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2143"
            lxc 20180106131213.591 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2146 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2146 0 100000 65536 65536 0 1
            lxc 20180106131213.591 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2146"
            lxc 20180106131213.592 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2149 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2149 0 100000 65536 65536 0 1
            lxc 20180106131213.592 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2149"
            lxc 20180106131213.594 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2152 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2152 0 100000 65536 65536 0 1
            lxc 20180106131213.594 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2152"
            lxc 20180106131213.595 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2155 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2155 0 100000 65536 65536 0 1
            lxc 20180106131213.595 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2155"
            lxc 20180106131213.596 ERROR    lxc_conf - conf.c:lxc_map_ids:2736 - newuidmap failed to write mapping "newuidmap: Target process 2158 is owned by a different user: uid:0 pw_uid:0 st_uid": newuidmap 2158 0 100000 65536 65536 0 1
            lxc 20180106131213.596 ERROR    lxc_conf - conf.c:userns_exec_1:3824 - error setting up {g,u}id mappings for child process "2158"
            lxc 20180106131213.596 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
            lxc 20180106131213.596 WARN     lxc_monitor - monitor.c:lxc_monitor_fifo_send:111 - Failed to open fifo to send message: No such file or directory.
```

 - [ ] Container configuration (`lxc config show NAME --expanded`)

```
architecture: x86_64
config:
  image.architecture: amd64
  image.description: Alpine 3.7 amd64 (20180105_17:50)
  image.os: Alpine
  image.release: "3.7"
  image.serial: "20180105_17:50"
  volatile.base_image: 3d5bbb978019fb3fd6bb486f2c239d4c3aa441db26fdc68e3c20dcbdbd4e8994
  volatile.idmap.base: "0"
  volatile.idmap.next: '[{"Isuid":true,"Isgid":false,"Hostid":100000,"Nsid":0,"Maprange":65536},{"Isuid":false,"Isgid":true,"Hostid":100000,"Nsid":0,"Maprange":65536}]'
  volatile.last_state.idmap: '[{"Isuid":true,"Isgid":false,"Hostid":100000,"Nsid":0,"Maprange":65536},{"Isuid":false,"Isgid":true,"Hostid":100000,"Nsid":0,"Maprange":65536}]'
  volatile.last_state.power: STOPPED
devices:
  root:
    path: /
    pool: default
    type: disk
ephemeral: false
profiles:
- default
stateful: false
description: ""
```

 - [ ] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log)

Not present in Alpine

 - [ ] Output of the client with --debug

```
DBUG[01-06|13:38:43] Connecting to a local LXD over a Unix socket 
DBUG[01-06|13:38:43] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0
DBUG[01-06|13:38:43] Got response struct from LXD 
DBUG[01-06|13:38:43] 
	{
		"config": {},
		"api_extensions": [
			"storage_zfs_remove_snapshots",
			"container_host_shutdown_timeout",
			"container_stop_priority",
			"container_syscall_filtering",
			"auth_pki",
			"container_last_used_at",
			"etag",
			"patch",
			"usb_devices",
			"https_allowed_credentials",
			"image_compression_algorithm",
			"directory_manipulation",
			"container_cpu_time",
			"storage_zfs_use_refquota",
			"storage_lvm_mount_options",
			"network",
			"profile_usedby",
			"container_push",
			"container_exec_recording",
			"certificate_update",
			"container_exec_signal_handling",
			"gpu_devices",
			"container_image_properties",
			"migration_progress",
			"id_map",
			"network_firewall_filtering",
			"network_routes",
			"storage",
			"file_delete",
			"file_append",
			"network_dhcp_expiry",
			"storage_lvm_vg_rename",
			"storage_lvm_thinpool_rename",
			"network_vlan",
			"image_create_aliases",
			"container_stateless_copy",
			"container_only_migration",
			"storage_zfs_clone_copy",
			"unix_device_rename",
			"storage_lvm_use_thinpool",
			"storage_rsync_bwlimit",
			"network_vxlan_interface",
			"storage_btrfs_mount_options",
			"entity_description",
			"image_force_refresh",
			"storage_lvm_lv_resizing",
			"id_map_base",
			"file_symlinks",
			"container_push_target",
			"network_vlan_physical",
			"storage_images_delete",
			"container_edit_metadata",
			"container_snapshot_stateful_migration",
			"storage_driver_ceph",
			"storage_ceph_user_name",
			"resource_limits",
			"storage_volatile_initial_source",
			"storage_ceph_force_osd_reuse",
			"storage_block_filesystem_btrfs",
			"resources",
			"kernel_limits",
			"storage_api_volume_rename",
			"macaroon_authentication",
			"network_sriov",
			"console",
			"restrict_devlxd",
			"migration_pre_copy",
			"infiniband",
			"maas_network"
		],
		"api_status": "stable",
		"api_version": "1.0",
		"auth": "trusted",
		"public": false,
		"auth_methods": [
			"tls"
		],
		"environment": {
			"addresses": [],
			"architectures": [
				"x86_64",
				"i686"
			],
			"certificate": "-----BEGIN CERTIFICATE-----XXX-----END CERTIFICATE-----\n",
			"certificate_fingerprint": "XXX",
			"driver": "lxc",
			"driver_version": "2.1.1",
			"kernel": "Linux",
			"kernel_architecture": "x86_64",
			"kernel_version": "4.9.73-0-virthardened",
			"server": "lxd",
			"server_pid": 1979,
			"server_version": "2.21",
			"storage": "dir",
			"storage_version": "1"
		}
	} 
DBUG[01-06|13:38:43] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/containers/settled-quagga
DBUG[01-06|13:38:43] Got response struct from LXD 
DBUG[01-06|13:38:43] 
	{
		"architecture": "x86_64",
		"config": {
			"image.architecture": "amd64",
			"image.description": "Alpine 3.7 amd64 (20180105_17:50)",
			"image.os": "Alpine",
			"image.release": "3.7",
			"image.serial": "20180105_17:50",
			"volatile.base_image": "3d5bbb978019fb3fd6bb486f2c239d4c3aa441db26fdc68e3c20dcbdbd4e8994",
			"volatile.idmap.base": "0",
			"volatile.idmap.next": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]",
			"volatile.last_state.idmap": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]",
			"volatile.last_state.power": "STOPPED"
		},
		"devices": {},
		"ephemeral": false,
		"profiles": [
			"default"
		],
		"stateful": false,
		"description": "",
		"created_at": "2018-01-06T13:09:12Z",
		"expanded_config": {
			"image.architecture": "amd64",
			"image.description": "Alpine 3.7 amd64 (20180105_17:50)",
			"image.os": "Alpine",
			"image.release": "3.7",
			"image.serial": "20180105_17:50",
			"volatile.base_image": "3d5bbb978019fb3fd6bb486f2c239d4c3aa441db26fdc68e3c20dcbdbd4e8994",
			"volatile.idmap.base": "0",
			"volatile.idmap.next": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]",
			"volatile.last_state.idmap": "[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]",
			"volatile.last_state.power": "STOPPED"
		},
		"expanded_devices": {
			"root": {
				"path": "/",
				"pool": "default",
				"type": "disk"
			}
		},
		"name": "settled-quagga",
		"status": "Stopped",
		"status_code": 102,
		"last_used_at": "2018-01-06T13:12:13.532352329Z"
	} 
DBUG[01-06|13:38:43] Connected to the websocket 
DBUG[01-06|13:38:43] Sending request to LXD                   etag= method=PUT url=http://unix.socket/1.0/containers/settled-quagga/state
DBUG[01-06|13:38:43] 
	{
		"action": "start",
		"timeout": 0,
		"force": false,
		"stateful": false
	} 
DBUG[01-06|13:38:43] Got operation from LXD 
DBUG[01-06|13:38:43] 
	{
		"id": "e3f98bb4-d9f7-4b5b-af4e-a856ff977414",
		"class": "task",
		"created_at": "2018-01-06T13:38:43.167556997Z",
		"updated_at": "2018-01-06T13:38:43.167556997Z",
		"status": "Running",
		"status_code": 103,
		"resources": {
			"containers": [
				"/1.0/containers/settled-quagga"
			]
		},
		"metadata": null,
		"may_cancel": false,
		"err": ""
	} 
DBUG[01-06|13:38:43] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/operations/e3f98bb4-d9f7-4b5b-af4e-a856ff977414
DBUG[01-06|13:38:43] Got response struct from LXD 
DBUG[01-06|13:38:43] 
	{
		"id": "e3f98bb4-d9f7-4b5b-af4e-a856ff977414",
		"class": "task",
		"created_at": "2018-01-06T13:38:43.167556997Z",
		"updated_at": "2018-01-06T13:38:43.167556997Z",
		"status": "Running",
		"status_code": 103,
		"resources": {
			"containers": [
				"/1.0/containers/settled-quagga"
			]
		},
		"metadata": null,
		"may_cancel": false,
		"err": ""
	} 


 - [ ] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)

metadata:
  context: {}
  level: dbug
  message: 'New events listener: 86281d30-54c4-4c57-85b0-01731f6be126'
timestamp: 2018-01-06T13:41:29.200206653Z
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.588609595Z
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0/containers/settled-quagga
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.595228276Z
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0/events
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.601461131Z
type: logging


metadata:
  context: {}
  level: dbug
  message: 'New events listener: 8688e5d5-2d75-4e7b-aec0-41b57eb465fb'
timestamp: 2018-01-06T13:43:08.602189625Z
type: logging


metadata:
  context:
    ip: '@'
    method: PUT
    url: /1.0/containers/settled-quagga/state
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.606065161Z
type: logging


metadata:
  context: {}
  level: dbug
  message: 'New task operation: ce12b6bb-5bd9-4845-adf5-2cde072dd7fb'
timestamp: 2018-01-06T13:43:08.606661892Z
type: logging


metadata:
  class: task
  created_at: 2018-01-06T13:43:08.606652356Z
  err: ""
  id: ce12b6bb-5bd9-4845-adf5-2cde072dd7fb
  may_cancel: false
  metadata: null
  resources:
    containers:
    - /1.0/containers/settled-quagga
  status: Pending
  status_code: 105
  updated_at: 2018-01-06T13:43:08.606652356Z
timestamp: 2018-01-06T13:43:08.606684526Z
type: operation


metadata:
  context: {}
  level: dbug
  message: 'Started task operation: ce12b6bb-5bd9-4845-adf5-2cde072dd7fb'
timestamp: 2018-01-06T13:43:08.606708482Z
type: logging


metadata:
  class: task
  created_at: 2018-01-06T13:43:08.606652356Z
  err: ""
  id: ce12b6bb-5bd9-4845-adf5-2cde072dd7fb
  may_cancel: false
  metadata: null
  resources:
    containers:
    - /1.0/containers/settled-quagga
  status: Running
  status_code: 103
  updated_at: 2018-01-06T13:43:08.606652356Z
timestamp: 2018-01-06T13:43:08.606720483Z
type: operation


metadata:
  context: {}
  level: dbug
  message: Initializing a DIR driver.
timestamp: 2018-01-06T13:43:08.613395507Z
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0/operations/ce12b6bb-5bd9-4845-adf5-2cde072dd7fb
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.615859938Z
type: logging


metadata:
  context:
    action: start
    created: 2018-01-06 13:09:12 +0000 UTC
    ephemeral: "false"
    name: settled-quagga
    stateful: "false"
    used: 2018-01-06 13:38:43.193058629 +0000 UTC
  level: info
  message: Starting container
timestamp: 2018-01-06T13:43:08.639643698Z
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.671019663Z
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /internal/containers/5/onstart
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.686295895Z
type: logging


metadata:
  context: {}
  level: dbug
  message: Initializing a DIR driver.
timestamp: 2018-01-06T13:43:08.689853195Z
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Scheduler: container settled-quagga started: re-balancing'
timestamp: 2018-01-06T13:43:08.695803706Z
type: logging


metadata:
  class: task
  created_at: 2018-01-06T13:43:08.606652356Z
  err: 'Failed to run: /usr/sbin/lxd forkstart settled-quagga /var/lib/lxd/containers
    /var/log/lxd/settled-quagga/lxc.conf: '
  id: ce12b6bb-5bd9-4845-adf5-2cde072dd7fb
  may_cancel: false
  metadata: null
  resources:
    containers:
    - /1.0/containers/settled-quagga
  status: Failure
  status_code: 400
  updated_at: 2018-01-06T13:43:08.606652356Z
timestamp: 2018-01-06T13:43:08.75424109Z
type: operation


metadata:
  context:
    action: start
    created: 2018-01-06 13:09:12 +0000 UTC
    ephemeral: "false"
    name: settled-quagga
    stateful: "false"
    used: 2018-01-06 13:38:43.193058629 +0000 UTC
  level: eror
  message: Failed starting container
timestamp: 2018-01-06T13:43:08.75418129Z
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Failure for task operation: ce12b6bb-5bd9-4845-adf5-2cde072dd7fb: Failed
    to run: /usr/sbin/lxd forkstart settled-quagga /var/lib/lxd/containers /var/log/lxd/settled-quagga/lxc.conf: '
timestamp: 2018-01-06T13:43:08.754224451Z
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.770338432Z
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Disconnected events listener: 8688e5d5-2d75-4e7b-aec0-41b57eb465fb'
timestamp: 2018-01-06T13:43:08.775226942Z
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /internal/containers/5/onstop?target=stop
  level: dbug
  message: handling
timestamp: 2018-01-06T13:43:08.786960896Z
type: logging


metadata:
  context: {}
  level: dbug
  message: Initializing a DIR driver.
timestamp: 2018-01-06T13:43:08.788766291Z
type: logging


metadata:
  context:
    action: stop
    created: 2018-01-06 13:09:12 +0000 UTC
    ephemeral: "false"
    name: settled-quagga
    stateful: "false"
    used: 2018-01-06 13:43:08.630569126 +0000 UTC
  level: info
  message: Container initiated stop
timestamp: 2018-01-06T13:43:08.78880881Z
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Scheduler: container settled-quagga stopped: re-balancing'
timestamp: 2018-01-06T13:43:09.699008897Z
type: logging
```

**Comment by stgraber** [comment]
Looks like there's something wrong going on with newuidmap/newgidmap on Alpine, those are usually provided by the shadow package and seem confused by process ownership in this case, causing the container startup failure.

Those helpers aren't actually required for LXD containers, so you could make them non-executable or move them out of the way, which would cause LXC to go straight to the kernel, avoiding the problem.

    chmod -x /usr/bin/newuidmap
    chmod -x /usr/bin/newgidmap

Should be enough. In some distros, those come from a separate package, which you could then just remove at least until whatever the issue is resolved in shadow.

**Comment by stgraber** [comment]
@brauner @hallyn does that newuidmap/newgidmap bug ring a bell? (I remember seeing this a while back on other distros)

**Comment by brauner** [comment]
On Jan 8, 2018 07:51, "Stéphane Graber" <notifications@github.com> wrote:

@brauner <https://github.com/brauner> @hallyn <https://github.com/hallyn>
does that newuidmap/newgidmap bug ring a bell? (I remember seeing this a
while back on other distros)


I vote we just always try to write directly when we are root and only call
newuidmap when we are unpriviliged.


—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/lxc/lxd/issues/4141#issuecomment-355893668>, or mute
the thread

**Comment by stgraber** [comment]
I think our current logic is perfectly sane, we use newuidmap/newgidmap if available so that the system administrator can properly configure what user can use what set of uids/gids. But if there is a bug in newuidmap/newgidmap in the checks it runs against the target process then this should get fixed.

**Comment by brauner** [comment]
There's no benefit in not trying to write to the files directly when we are root and it also saves us from spawning a separate process, waiting on it and capturing its output.

**Comment by brauner** [comment]
For the problem itself, this is caused by grsecurity.

**Comment by brauner** [comment]
Fwiw, this grsecurity example is where not going through a separate process will actually let the user start a container since grsecurity blocks writing newuidmap for a process of which you are not the parent.

**Comment by brauner** [comment]
@lePereT, can you rename the `new{g,u}idmap` binaries temporarily to some random name and then try to start a container again, please.

**Comment by stgraber** [comment]
Ah, grsec, that's what I couldn't remember and the kernel name kinda hints at it.
Closing since that's really a grsec/shadow issue, not something LXD can do anything about.
Moving newuidmap/newgidmap out of the way, uninstalling them or making them non-executable should all work as a workaround here, though grsec is likely to interfere in other ways too...

@brauner what we loose with writing directly to the kernel when we're root is the ability to have the sysadmin slice the uid/gid range and make sure that nothing will step across those boundaries as the kernel will very happily let us map whatever we want regardless of what the administrator configured, possibly interfering with ranges that have been delegated to other users.

**Comment by BlueMax** [comment]
I'm in the same boat. It works indeed when disabling the two binaries. 
The 1st container gets uid 100.000, but 2nd, 3rd and 4th get unified uid 1.000.000. Can we configure/workaround that somehow (without recompiling)?
